File: Examples/CallGraph/Swift/Services.swift
/// Places an order by charging payment and persisting it.
class OrderService {
    let payment: PaymentService
    let repository: OrderRepository

    init(payment: PaymentService, repository: OrderRepository) {
        self.payment = payment
        self.repository = repository
    }

    func place(_ order: Order) {
        payment.charge(order)
        repository.save(order)
    }
}

class PaymentService {
    func charge(_ order: Order) {}
}

class OrderRepository {
    func save(_ order: Order) {}
}

File: Examples/CallGraphDiff/Python/Before/services.py
from orders import Order


class OrderService:
    """Places an order by charging payment and persisting it."""

    def __init__(self, payment: "PaymentService", repository: "OrderRepository"):
        self.payment: PaymentService = payment
        self.repository: OrderRepository = repository

    def place(self, order: Order) -> None:
        self.payment.charge(order)


class PaymentService:
    def charge(self, order: Order) -> None:
        pass


class OrderRepository:
    def save(self, order: Order) -> None:
        pass
